File: SW-Expert-Academy/15868.cpp
#include <bits/stdc++.h>

using namespace std;

char arr[5][5];

int main(){
    ios::sync_with_stdio(false); cin.tie(nullptr);

    int t;
    cin >> t;
    for(int x=1; x<=t; x++){
        int n, m; cin >> n >> m;
        string first;
        cin >> first;

        string ret = "yes";
        
        for(int i=1; i<n; i++){
            string arr;
            cin >> arr;

            int a, b;
            a = int(first[0]) - 48;

            if(int(arr[0])-48) b = !a;
            else b = a;

            for(int j=1; j<m; j++){
                a = int(first[j]) - 48;
                if((b^a) != (int(arr[j])-48)){
                    ret = "no";
                    break;
                }
            }
        }

        cout << "#" << x << " " << ret << endl;
        
    }
    
}

File: SW-Expert-Academy/15868.py
import sys

t = int(sys.stdin.readline().rstrip())

for x in range(1, t+1):
    n, m = map(int, sys.stdin.readline().rstrip().split())
    
    first = sys.stdin.readline().rstrip()

    ret = 'yes'
    
    for _ in range(1, n):
        arr = sys.stdin.readline().rstrip()
        
        a = int(first[0])
        
        if bool(int(arr[0])):
            b = int(not a)
        else:
            b = a

        for i in range(1, m):
            a = int(first[i])

            if b ^ a != int(arr[i]):
                ret = 'no'
                break
    
    print('#'+str(x), ret)